src/encryptdecrypt/EncryptorDecryptor.java

```diff
@@ -0,0 +1,98 @@
+package encryptdecrypt;
+
+import encryptdecrypt.strategy.Algorithm;
+import encryptdecrypt.strategy.ShiftAlgorithm;
+import encryptdecrypt.strategy.UnicodeAlgorithm;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+public class EncryptorDecryptor {
+
+    private Algorithm coder;
+    private String mode, data, out, in;
+    private int key;
+
+    private void setCoder(String type) {
+        switch (type) {
+            case "shift":
+                coder = new ShiftAlgorithm();
+                break;
+            case "unicode":
+                coder = new UnicodeAlgorithm();
+                break;
+        }
+    }
+
+    public EncryptorDecryptor() {
+        coder = new ShiftAlgorithm();
+        mode = "enc";
+        data = "";
+        key = 0;
+    }
+
+    public void start(String[] args) {
+        parseArgs(args);
+
+        prepareData();
+        processData();
+        outputData();
+    }
+
+    private void prepareData() {
+        if (data.isEmpty() && in != null) {
+            try {
+                data = readFileAsString(in);
+            } catch (IOException e) {
+                System.out.println("Error");
+            }
+        }
+    }
+
+    private void processData() {
+        key = (mode.equals("enc") ? key : -key);
+
+        data = coder.crypt(data, key);
+    }
+
+    private void outputData() {
+        if (out != null) {
+            try (FileWriter writer = new FileWriter(out)) {
+                writer.write(data);
+            } catch (IOException e) {
+                System.out.println("Error");
+            }
+        } else
+            System.out.println(data);
+    }
+
+    private void parseArgs(String[] args) {
```

Comment:
`EncryptorDecryptor` should not parse arguments

Reply:
Fixed